test/integration/profile_test.rb

```diff
@@ -133,6 +133,19 @@ def sign_out
     assert page.has_link?("@nick1", href: "https://twitter.com/nick1")
   end
 
+  test "adding X(formerly Twitter) username without filling in your password" do
+    sign_in
+    visit profile_path("nick1")
+
+    click_link "Edit Profile"
+    fill_in "user_twitter_username", with: "nick1twitter"
+    click_button "Update"
+        puts page.html
+
+
+    assert_content("please fill in your password")
```

Comment:
This may not be readable by capybara because it's browser generated.

What if you click submit, comment about what's happening (browser halted submit because password is required), then filled the password and submitted again and check that the new value is saved.

Reply:
Done! I had to write this as a system test rather than an integration test since the system test runs the test on actual chrome.